tests/integ/test_trial_component.py

```diff
@@ -23,6 +23,16 @@ def test_create_delete(trial_component_obj):
     assert trial_component_obj.trial_component_name
 
 
+def test_delete_with_disassociate(trial_component_with_disassociation_obj, sagemaker_boto_client):
+    assert trial_component_with_disassociation_obj.trial_component_name
+    trials = sagemaker_boto_client.list_trials(
+        TrialComponentName=trial_component_with_disassociation_obj.trial_component_name
+    )["TrialSummaries"]
+    assert len(trials) == 3
```

Comment:
seems like this is just testing that test data is setup as expected, not actual sdk behavior - remove?

Reply:
I was doing some checks to make sure the there are multiple trials associate with trial component. In line 134, it will sleep for 0.5 second then call trial_component_obj.delete(disassociate=True). This is how test_create_delete() test delete as well.